src/main/kotlin/com/dropbox/mobile/hypershard/Hypershard.kt

```diff
@@ -324,15 +357,21 @@ class RealHyperShard(
             }
 
             for (element in declaration.children) {
-                val ktClassBody = element as? KtClassBody ?: continue
+                val ktClassBody = element as? DefaultAstNode ?: continue
+                if (ktClassBody.description != "classBody") {
+                    continue
+                }
 
                 for (fn in ktClassBody.children) {
-                    val ktNamedFunction = fn as? KtNamedFunction ?: continue
+                    val ktNamedFunction = fn as? KlassDeclaration ?: continue
+                    if (ktNamedFunction.keyword != "fun") {
+                        continue
+                    }
                     // the fqName includes the test name with the full package name and test name
                     // all delimited by periods, this allows the test to be ran because we need
                     // the hashtag
-                    for (fnAnnotationEntry in ktNamedFunction.annotationEntries) {
-                        if (fnAnnotationEntry.shortName.toString() == "Test") {
+                    for (fnAnnotationEntry in ktNamedFunction.annotations) {
+                        if (fnAnnotationEntry.identifier.first().identifier == "Test") {
```

Comment:
should this be firstOrNull(), could identifier be an empty collection?

Reply:
Probably! Thanks